code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,182 @@
+
+/obj/item/device/nanoquikpay
+	name = "\improper NT Quik-Pay"
+	desc = "Swipe you're ID to make direct company purcheses"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "quikpay"
+	var/machine_id = ""
+	var/list/items = list()
+	var/sum = 0
+	var/access_code = 0
+	var/editmode = 0
+
+
+
+
+/obj/item/device/nanoquikpay/New()
+	..()
+	machine_id = "[station_name()] Quik-Pay #[SSeconomy.num_financial_terminals++]"
+	access_code = rand(1111,111111)
+	spawn(0)
```

Comment:
I don't necessarily see a reason why this should be spawn()-d? Though if there is a reason, use a timer.